### ISSUE TITLE
Migrate certificate handling to terraform?

### ISSUE THREAD

**Issue by hingstarne** [kz8s/tack#121]
Terraform itself can do the certificate handling, with the tls provider.
Are there any plans to migrate from cfssl and make to terraform tls provider based approach?
Would a pr regarding this be welcomed?

**Comment by wellsie** [comment]
Absolutely

**Comment by grncdr** [comment]
Hey, I don't know if this is still something you're interested in, but I've implemented a first version of it in https://github.com/talon-one/tack/commit/c9cdec43a9fbdc3ddce8f6658fae9ee392dbd773.

I simply generate all the private keys internally, but it would probably be better to accept private keys as top-level variables. If the approach doesn't look too bad, I will continue with removing `cfssl` from my fork and open a PR.

**Comment by alertedsnake** [comment]
@grncdr I looked at this briefly - there are two issues with doing this via passing the certs into the instance userdata - one, that that data is then available to anyone who can view it in the EC2 console or get access to the machine (metadata-service), and two, there is an upper limit of the size of the instance userdata.  I went down this path and hit the limit, and then realized the security issues.

So I'm using Terraform to create these, but storing them in S3 the way this was originally done.
